app/components/markdown/markdown_latex_block/index.tsx

```diff
@@ -0,0 +1,232 @@
+}
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => {
+    const codeVerticalPadding = Platform.select({
+        ios: 4,
+        android: 0,
+    });
+
+    return {
+        bottomSheet: {
+            flex: 1,
+        },
+        container: {
+            borderColor: changeOpacity(theme.centerChannelColor, 0.15),
+            borderRadius: 3,
+            borderWidth: StyleSheet.hairlineWidth,
+            flexDirection: 'row',
+            flex: 1,
+        },
+        rightColumn: {
+            flexDirection: 'column',
+            flex: 1,
+            paddingLeft: 6,
+            paddingVertical: 4,
+        },
+        code: {
+            flexDirection: 'row',
+            justifyContent: 'flex-start',
+            marginLeft: 5,
+            paddingVertical: codeVerticalPadding,
+        },
+        plusMoreLinesText: {
+            color: changeOpacity(theme.centerChannelColor, 0.4),
+            fontSize: 11,
+            marginTop: 2,
+        },
+        language: {
+            alignItems: 'center',
+            backgroundColor: theme.sidebarHeaderBg,
+            justifyContent: 'center',
+            opacity: 0.8,
+            padding: 6,
+            position: 'absolute',
+            right: 0,
+            top: 0,
+        },
+        languageText: {
+            color: theme.sidebarHeaderTextColor,
+            fontSize: 12,
+        },
+        errorText: {
+            fontSize: 14,
+            marginHorizontal: 5,
+            color: theme.errorTextColor,
+        },
+    };
+});
+
+const LatexCodeBlock = ({content, theme}: Props) => {
+    const intl = useIntl();
+    const insets = useSafeAreaInsets();
+    const managedConfig = useManagedConfig<ManagedConfig>();
+    const styles = getStyleSheet(theme);
+
+    const handlePress = preventDoubleTap(() => {
+        const screen = Screens.LATEX;
+        const passProps = {
+            content,
+        };
+
+        const languageDisplayName = getHighlightLanguageName('latex');
+        const title = intl.formatMessage({
+            id: 'mobile.routes.code',
+            defaultMessage: '{language} Code',
+        }, {
+            language: languageDisplayName,
+        });
+
+        Keyboard.dismiss();
+        requestAnimationFrame(() => {
+            goToScreen(screen, title, passProps);
+        });
+    });
+
+    const handleLongPress = useCallback(() => {
+        if (managedConfig?.copyAndPasteProtection !== 'true') {
+            const renderContent = () => {
+                return (
+                    <View
+                        testID='at_mention.bottom_sheet'
+                        style={styles.bottomSheet}
+                    >
+                        <SlideUpPanelItem
+                            icon='content-copy'
+                            onPress={() => {
+                                dismissBottomSheet();
+                                Clipboard.setString(content);
+                            }}
```

Comment:
The useCallback is not taking care of this memoization, so this will mean one new function on every render.

I am not sure of the implications in this context, since it is a `renderContent` function passed to the bottom sheet.